src/mpyl/steps/models.py

```diff
@@ -92,35 +93,41 @@ class RunProperties:
     console: ConsoleProperties
     """Settings for the console output"""
     stages: list[Stage]
-    """Stage definitions"""
+    """All stage definitions"""
+    projects: set[Project]
+    """All projects"""
+    run_plan: dict[Stage, set[Project]]
+    """Stages and projects for this run"""
 
     @staticmethod
     def for_local_run(
         config: dict,
+        run_plan: dict[Stage, set[Project]],
         revision: str,
         branch: Optional[str],
-        tag: Optional[str],
-        stages: list[dict],
+        tag: Optional[str] = None,
+        stages: Optional[list[dict]] = None,
```

Comment:
Why is this optional? Also what's the difference between None and [] in this case?